scripts/generateMetadata.js

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env node
+const chalk = require('chalk');
+const prettier = require('prettier');
+
+const reactDocs = require('react-docgen');
+const fs = require('fs');
+const path = require('path');
+
+const root = path.join(__dirname, '../');
+const docsPath = path.join(root, '/docs');
+
+function logError(message) {
+  // eslint-disable-next-line no-console
+  console.log(chalk.red(`❌  Error: ${message}`));
+}
+
+function logSuccess(message) {
+  // eslint-disable-next-line no-console
+  console.log(chalk.green(`✅ ${message}`));
+}
+
+async function docgen(filePath) {
+  const contents = await fs.promises.readFile(filePath, 'utf-8');
+
+  // Not all files have data to parse
+  try {
+    const parsed = reactDocs.parse(contents);
+
+    if (parsed.description) {
+      parsed.description = parsed.description
+        // Remove the first markdown link from the description so we don't link to the page itself
+        .replace(/\[(.*?)\][[(].*?[\])]/, '$1')
+        // Remove images from the description
+        .replace(/!\[(.*?)\][[(].*?[\])]/g, '');
+    }
+
+    return parsed;
+  } catch {
+    return null;
+  }
+}
+
+async function getFilesFromFolder(folder) {
+  const folderPath = path.join(root, folder);
+  const folderFiles = await fs.promises.readdir(folderPath);
+
+  const supportedFiles = [];
+
+  for (let i = 0; i < folderFiles.length; i += 1) {
+    // Filter no js files and test files
+    const fileName = folderFiles[i];
+    if (fileName.match(/\.(js)$/i) && !fileName.match(/\.(test\.js)$/i)) {
+      supportedFiles.push(path.join(folder, fileName));
+    }
+  }
+
+  return supportedFiles;
+}
+
+(async function generateComponentsMetadata() {
+  const folders = ['/packages/gestalt/src/'];
+  const files = [
```

Comment:
Can we avoid needing to hardcode these paths?

Reply:
I added the function to recursively check the '/packages/gestalt/src' folder. I added an array for 'extraFiles' (we only have one right now).